src/RequestsLibrary/RequestsKeywords.py

```diff
@@ -48,9 +49,16 @@ def _common_request(self, method, session, uri, **kwargs):
 
         files = kwargs.get("files", {}) or {}
         data = kwargs.get("data", []) or []
-        files_descriptor_to_close = filter(
-            is_file_descriptor, list(files.values()) + [data]
-        )
+
```

Comment:
at this point I would extract the logic to close file descriptor in a dedicated function and try to cleanup this part

Reply:
I refactored the closing of file descriptors into its own static method.